src/affordance_entity.js

```diff
@@ -0,0 +1,117 @@
+//this file will contain the entity class which keeps all info about the affordance and some actions that manipulate the affordance
+import { addMiaIcon, addLoader, deleteLoader, addInfoIcon, addFailed } from "./node_modifications.js";
+import { createEmptyStore, storeSize , getLinkedDataNQuads, addDataToStore} from "./linked_data_store.js";
+import { getBoundryInfo, getInfoPopup } from "./info_extraction.js";
+import Popup from "./popup.js";
+
+export default class AffordanceEntity {
+    constructor(affordance) {
+        this.uri = affordance.href;
+        this.id = 'id-' + Math.random().toString(36).slice(2, 11); // todo: check if this is really needed in the end
+        this.node = affordance;
+        this.store = createEmptyStore();
+        this.checked = false;
+        this.icon = null;
+
+        //add the id to the node as the mia_entity_id attribute
+        this.node.setAttribute('mia_entity_id', this.id);
+
+        //do check if uri is from marine regions or marine info or any other known source with linked data
+        this.checkUri();
+        //perform span manipulation based on info from the initial class creation
+        this.NodeManipulations();
```

Comment:
odd uppercase method name `NodeManipulations` -- suggests static which would be awkward

also `manipulations` is generic `doing` `function` `process` `method` all are not adding clear semantics about the actual purpose of this method -- so unhelpfully abstract